vnext/local-cli/generator-common/index.js

```diff
@@ -224,9 +225,17 @@ function copyAndReplace(
     // Text file
     const srcPermissions = fs.statSync(srcPath).mode;
     let content = fs.readFileSync(srcPath, 'utf8');
-    Object.keys(replacements).forEach(regex => {
-      content = content.replace(new RegExp(regex, 'g'), replacements[regex]);
-    });
+
+    if (replacements.useMustache) {
+      content = mustache.render(content, replacements);
+      (replacements.regExpPatternsToRemove || []).forEach(regexPattern => {
```

Comment:
I tried to handle the `// clang-format off` and `// clang-format on` cases with template logic, but I could not get a satisfactory line that clang-format was happy with. It would always mangle the conditional line before the off command and the closing conditional after the on command. It was pretty messy... 
I therefore opted for a 1-off removal.

I considered a waterfall model as well where when mustach is on, and a tempaltes fields is set it would set the replacements to that value (else undefined) and it would run the old code. In the end I felt that that code would be a lot harder to read and understand vs the simple foreach and replace slighlty duplicated.